File: api-net/Controllers/AtfController.cs
using System.Text.Json;
using LiftOtonom.Api.Data;
using LiftOtonom.Api.Models;
using LiftOtonom.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace LiftOtonom.Api.Controllers;

[ApiController]
[Authorize]
[Route("api/v1/atf")]
public class AtfController(AppDbContext db) : ControllerBase
{
    public record AtfDto(long? CustomerId, long? BuildingId, JsonElement? FormData, string? Status, JsonElement? Attachments);

    [HttpGet]
    public async Task<IActionResult> Index([FromQuery(Name = "per_page")] int perPage = 25, [FromQuery] int page = 1) =>
        Ok(await db.AtfForms.OrderByDescending(a => a.Id).ToPagedAsync(page, perPage));

    [HttpPost]
    public async Task<IActionResult> Store(AtfDto dto)
    {
        var now = DateTime.UtcNow;
        var a = new AtfForm
        {
            TenantId = db.CurrentTenantId!.Value, CustomerId = dto.CustomerId, BuildingId = dto.BuildingId,
            FormData = dto.FormData?.GetRawText() ?? "{}", Status = dto.Status ?? "draft",
            Attachments = dto.Attachments?.GetRawText() ?? "[]",
            CreatedBy = long.Parse(User.FindFirst("uid")!.Value), CreatedAt = now, UpdatedAt = now,
        };
        db.AtfForms.Add(a);
        await db.SaveChangesAsync();
        return StatusCode(201, a);
    }

    [HttpGet("{id:long}")]
    public async Task<IActionResult> Show(long id) =>
        Ok(await db.AtfForms.FirstOrDefaultAsync(x => x.Id == id) ?? throw new ApiException(404, "ATF bulunamadı."));

    [HttpPut("{id:long}")]
    public async Task<IActionResult> Update(long id, AtfDto dto)
    {
        var a = await db.AtfForms.FirstOrDefaultAsync(x => x.Id == id) ?? throw new ApiException(404, "ATF bulunamadı.");
        if (dto.FormData is { } fd) a.FormData = fd.GetRawText();
        if (dto.Attachments is { } at) a.Attachments = at.GetRawText();
        if (dto.Status != null) a.Status = dto.Status;
        a.UpdatedAt = DateTime.UtcNow;
        await db.SaveChangesAsync();
        return Ok(a);
    }

    [HttpDelete("{id:long}")]
    public async Task<IActionResult> Destroy(long id)
    {
        var a = await db.AtfForms.FirstOrDefaultAsync(x => x.Id == id) ?? throw new ApiException(404, "ATF bulunamadı.");
        a.DeletedAt = DateTime.UtcNow;
        await db.SaveChangesAsync();
        return Ok(new { message = "ATF silindi." });
    }
}

File: api/app/Http/Controllers/Api/AtfController.php
<?php
// app/Http/Controllers/Api/AtfController.php

namespace App\Http\Controllers\Api;

use App\Http\Controllers\Controller;
use App\Models\AtfForm;
use Illuminate\Http\Request;

class AtfController extends Controller
{
    public function index(Request $request)
    {
        return AtfForm::query()->orderByDesc('id')
            ->paginate(min((int) $request->integer('per_page', 25), 100));
    }

    public function store(Request $request)
    {
        $data = $request->validate([
            'customer_id' => ['nullable', 'integer', 'exists:customers,id'],
            'building_id' => ['nullable', 'integer', 'exists:buildings,id'],
            'form_data'   => ['nullable', 'array'],
            'attachments' => ['nullable', 'array'],
            'status'      => ['nullable', 'string', 'max:50'],
        ]);
        $data['created_by'] = $request->user()->id;

        return response()->json(AtfForm::create($data)->fresh(), 201);
    }

    public function show(AtfForm $atf)
    {
        return $atf;
    }

    public function update(Request $request, AtfForm $atf)
    {
        $atf->update($request->validate([
            'form_data'   => ['nullable', 'array'],
            'attachments' => ['nullable', 'array'],
            'status'      => ['nullable', 'string', 'max:50'],
        ]));

        return $atf->fresh();
    }

    public function destroy(AtfForm $atf)
    {
        $atf->delete();

        return response()->json(['message' => 'ATF silindi.']);
    }
}
